### PR TITLE
[Validator] Remove support of boolean value for the checkDNS option

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | /
| License       | MIT
| Doc PR        | /

Initial PR https://github.com/symfony/symfony/pull/23076

### CODE REVIEW

**Review by xabbuh — CHANGES_REQUESTED** [review]
please update the changelog too

**Comment by maidmaid** [comment]
Updated.

File: src/Symfony/Component/Validator/CHANGELOG.md [hunk 1]
```diff
@@ -8,13 +8,14 @@ CHANGELOG
...
  * removed the `AbstractConstraintValidatorTest` class
+ * removed the support for setting the `checkDNS` option of the `Url` constraint to `true`
 
 3.4.0
 -----
 
  * not setting the `strict` option of the `Choice` constraint to `true` is
    deprecated and will throw an exception in Symfony 4.0
- * setting the `checkDNS` option of the `Url` constraint to `true` is deprecated in favor of constant values and will throw an exception in Symfony 4.0
+ * setting the `checkDNS` option of the `Url` constraint to `true` is deprecated in favor of `Url::CHECK_DNS_TYPE_*` constants values and will throw an exception in Symfony 4.0
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/CHANGELOG.md]: this should be done in the `3.4` branch

File: src/Symfony/Component/Validator/CHANGELOG.md [hunk 2]
```diff
@@ -8,13 +8,14 @@ CHANGELOG
    is not supported anymore.
  * removed the `DateTimeValidator::PATTERN` constant
  * removed the `AbstractConstraintValidatorTest` class
+ * removed the support for setting the `checkDNS` option of the `Url` constraint to `true`
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/CHANGELOG.md]: removed support [...]

File: UPGRADE-4.0.md [hunk 3]
```diff
@@ -569,6 +569,21 @@ Validator
    changed to `true` as of 4.0. If you need the previous behaviour ensure to
    set the option to `false`.
 
+ * Setting the `checkDNS` option of the `Url` constraint to `true` is deprecated
```
Comment (xabbuh) [inline UPGRADE-4.0.md]: support ist dropped in 4.0, not deprecated

File: src/Symfony/Component/Validator/CHANGELOG.md (hunk 1 above)
Comment (maidmaid) [inline src/Symfony/Component/Validator/CHANGELOG.md]: Removed. Is this change worth opening a new PR?

File: src/Symfony/Component/Validator/CHANGELOG.md (hunk 2 above)
Comment (maidmaid) [inline src/Symfony/Component/Validator/CHANGELOG.md]: Updated

File: UPGRADE-4.0.md (hunk 3 above)
Comment (maidmaid) [inline UPGRADE-4.0.md]: Oops, it's updated now.

File: src/Symfony/Component/Validator/CHANGELOG.md (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/Validator/CHANGELOG.md]: see ac107ba6b253af9d217cf511ceb10f1d47d91ed8

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @maidmaid.

**Comment by javiereguiluz** [comment]
I don't like this change being merged. I'd prefer to keep the current behavior (when using `true`, you mean `ANY`) because that's the common case. In fact, the example you showed with the before/after differences is nice ... but that's PHP. However, most people define these constraints with annotations or YAML files. It won't be that easy for them :(

```php
/** @Assert\Url(checkDNS = true) */
protected $bioUrl;
```

```yaml
AppBundle\Entity\Author:
    properties:
        bioUrl:
            # before
            - Url: { checkDNS: true }

            # after ("the right way")
            - Url: { checkDNS: !php/const:Symfony\Component\Validator\Constraints\Url::CHECK_DNS_TYPE_ANY }

            # after (with "magic" strings)
            - Url: { checkDNS: 'ANY' }
```

By the way, `checkDNS: 'ANY'`, may look like, "anything" will work for me, I don't care about this value.

**Comment by fabpot** [comment]
I tend to agree with Javier but would go even one step further and remove this option altogether. It's not really that useful IMHO to check DNS.

**Comment by maidmaid** [comment]
@javiereguiluz Only `Url::CHECK_DNS_TYPE_*` constants values are accepted. ``checkDNS: 'ANYTHING'`` [throws an exception]( https://github.com/symfony/symfony/pull/23076/files#diff-c152f2b5c9ff3db02b5270f3e6be2995R96). Thus, ``checkDNS: 'ANY'`` is maybe to recommend in annotations/YAML case.

**Pull request merged**